article.md

```diff
@@ -0,0 +1,612 @@
+# So you want to use types with Sequelize?
+
+Once a team first takes the leap into the world of types with Typescript, it seems there's no going back. Typescript has become common to see in front-end applications, so why aren't we using all those juicy compile-time type checks on the server? At Vivacity, we've adopted Typescript on *both* the front- and back-end for new projects. We're currently building a new data set management tool using Typescript, Express and Sequelize. This guide will step you through how to set this up so you can create, read, update and delete without ever getting a type wrong.
+
+---
+
+## Plan
+
+1. Explain the website and schema we are going to make
+2. Setup: First, set up a default project with simple stuff.
+3. Api design: Set up project to use controllers.
+4. talk about how we're going to have each model have its own file. Except, rather than making it a default export, we're going to export a ModelFactory function, and two interfaces, ModelAttributes and ModelInstance. We avoid default exports because it makes type inference worse in typescript.
+5. Model loader: We then have an index.ts file, similar to with js, that loads them up. Except, we also need a DbInterface type. It does this and doesn't load all files in a directory, because otherwise you'd lose type inference
+6. Model definitions: Explain ModelAttrs, ModelInstance, SequelizeAttrs
+7. Associations and the mixins
+8. Glorious final result: writing api routes with lots of beautiful type inference (screenshot of intellisense)
+
+### Plan:  Schema
+
+- Users: firstName, lastName
+- Posts: AuthorId, title, text, category: enum("tech", "croissants", "techno")
+- Comments: PostId, AuthorId, text
+- PostUpvotes: PostId, UpvoterId
+
+Users (Author) 1:m Posts  
+Users (Author) 1:m Comments  
+Posts 1:m Comments  
+User (Upvoter) n:m Comments
+
+---
+
+## Let's get started
+
+We're about to embark on a rollercoaster adventure. At first, adding Typescript will seem like nothing but a cumbersome waste of time. Hours lost to just littering your Sequelize models with a slew of verbose type annotations. But, peservere! By the time we reach the end, your models will be so well typed that your api routes will practically be writing themselves. Here's a sneak peak.
```

Comment:
This part seems a bit over the top to me 😄 - devs will usually look for the "meat and bones", so I'd keep the intro short and snazzy. Should we also reference the existing material by adding something like "There are _some_ guides out there, such as [this useful guide](https://michalzalecki.com/using-sequelize-with-typescript/) by Michal Zalecki. While that can get you start, it doesn't tell you how to x, y, z / does not deal with associations, blah blah"

Reply:
I guess it might also help to say something along the lines of "usually adding types is a matter of adding a `@types` package, but for Sequelize this is harder because of associations, blah, blah. 

I recall `@types/sequelize` being somewhat outdated (or for v3?) - was that the case? If so, worth mentioning

Reply:
also typo -> persevere

Reply:
I agree it's pretty over the top and isn't typical to have such a paragraph in an article that serves as just a technical guide (e.g. the Zalecki article).

I think, for me, the idea behind having a paragraph like this stems from the idea of this article serving a dual purpose. As a technical guide and, as a Vivacity blog. So, the occasional 'chatter' is maybe permissible? This would be on the basis that we want it to also serve for a more general reader who is just trying to find out more about the company or the company's team / careers. My idea's that some text like this makes the company's employyes seem like lovely approachable and hopefully gives an impression of a friendly office.

What are your thoughts? 😄

Reply:
I'm definitely going to add the references to others and explain why its a bit harder than just adding the `@types` package. Thanks 💯